### PR TITLE
Changed default settings for SA1316 to PascalCase. 

### LINKED ISSUE #3019: The default tuple element names config should align with the new pascal case guidance
The BCL and language teams have decided on pascal case for tuple element names and have added an API to the BCL which sets the precedent. (https://github.com/dotnet/corefx/pull/35595, 
https://github.com/dotnet/corefx/issues/33553)

Under https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1316.md#rule-description:

> For example, the following code would produce a violation of this rule (with default settings):
> 
> ```csharp
> public class ExampleClass
> {
>   public (int ValueA, int ValueB) ExampleMethod() // SA1316
>   {
>   }
> }
> ```

This should not default to being a violation. I would even expect that camel case would be considered a violation as soon as this rule is enabled in a new project.

This rule does not exist in the latest release version, so maybe backwards compatibility won't be an issue.

### PR DESCRIPTION
**Pull request by vweijsters**
Added missing C# 8 tests for SA1316.

Fixes #3019

### CODE REVIEW

**Comment by sharwell** [comment]
Pending a resolution of discussion in #3019

File: StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/NamingRules/SA1316CSharp8UnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Test.CSharp8.NamingRules
+{
+    using StyleCop.Analyzers.Test.CSharp7.NamingRules;
+
+    public class SA1316CSharp8UnitTests : SA1316CSharp7UnitTests
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/NamingRules/SA1316CSharp8UnitTests.cs]: Was this just missed by accident?

File: documentation/Configuration.md [hunk 2]
```diff
@@ -330,7 +330,7 @@ The following example shows a settings file which requires tuple element names t
   "settings": {
     "namingRules": {
       "includeInferredTupleElementNames": true,
-      "tupleElementNameCasing" : "PascalCase"
+      "tupleElementNameCasing" : "camelCase"
```
Comment (sharwell) [inline documentation/Configuration.md]: :bulb: make sure the preceding text is consistent with this change (I can't see the full text on this small screen)

File: StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/NamingRules/SA1316CSharp8UnitTests.cs (hunk 1 above)
Comment (vweijsters) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/NamingRules/SA1316CSharp8UnitTests.cs]: Yes, as SA1316 doesn't have any tests for CSharp 6, I probably missed it when adding the C# 8 tests.

File: documentation/Configuration.md (hunk 2 above)
Comment (vweijsters) [reply documentation/Configuration.md]: Fixed it

**Pull request merged**